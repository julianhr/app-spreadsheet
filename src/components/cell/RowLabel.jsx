import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'


const Wrapper = styled.div`
  font-size: 12px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-right: 1px solid ${props => props.theme.colors.cell.borderDark};
  border-bottom: 1px solid ${props => props.theme.colors.cell.borderDark};
  background: ${props => props.theme.colors.cell.labelBkg};
`

function RowLabel({ label }) {
  return (
    <Wrapper
      data-row={label}
      className='row-label-width row-label-height'
    >
      {label}
    </Wrapper>
  )
}

RowLabel.propTypes = {
  label: PropTypes.number,
}

export default RowLabel
