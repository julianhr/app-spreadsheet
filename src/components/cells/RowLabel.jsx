import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'


const Wrapper = styled.div`
  font-size: 12px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-right: 1px solid ${props => props.theme.colors.cell.borderDark};
  border-bottom: 1px solid ${props => props.theme.colors.cell.borderDark};
  background: ${props => props.theme.colors.cell.labelBkg};
`

function RowLabel({ label }) {
  return (
    <td>
      <Wrapper
        className='row-label-width row-label-height'
      >
        {label}
      </Wrapper>
    </td>
  )
}

RowLabel.propTypes = {
  label: PropTypes.number,
}

export default RowLabel
