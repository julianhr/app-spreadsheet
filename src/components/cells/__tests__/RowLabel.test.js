import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import MockApp from '~/__tests__/__mocks__/MockApp'
import RowLabel from '../RowLabel'


const renderApp = (props) => {
  const wrapper = render(
    <MockApp>
      <table>
        <tr>
          <RowLabel {...props} />
        </tr>
      </table>
    </MockApp>
  )

  return wrapper
}

describe('RowLabel', () => {
  afterEach(cleanup)

  it('matches snapshot', () => {
    const props = {
      label: 1
    }

    renderApp(props)
    const el = document.querySelector('[data-row="1"]')
    expect(el).toMatchSnapshot()
  })
})
