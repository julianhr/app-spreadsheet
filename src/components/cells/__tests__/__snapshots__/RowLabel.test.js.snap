// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowLabel matches snapshot 1`] = `
<td>
  <div
    class="row-label-width row-label-height css-1cj8orw-Wrapper ewfjn6o0"
  >
    1
  </div>
</td>
`;
